Courses.py attempt 2

Tommy-Nguyen17 opened the pull request:
added type defs to functions

adds class with add, modify and delete methods for courses

note that there is no error checking in this current version

Tommy-Nguyen17 committed:
courses.py

Had to delete old branch to fix something but here's the courses.py base file. Still needs getters but setters are basically done

Just needs to change the initializer a little bit to use the JsonConfig class to get the Config object

Tommy-Nguyen17 committed:
Update to courses.py

update courses py with the add_courses function

Tommy-Nguyen17 committed:
Update courses.py

Tommy-Nguyen17 committed:
added header for function that deletes a course

Tommy-Nguyen17 committed:
implemented delete and modify courses functions

the functions for mod course and delete courses have been implemented

Tommy-Nguyen17 committed:
added type def to functions

Tommy-Nguyen17 committed:
Create example.json

ZikoGotGame commented on example.json:
```diff
@@ -0,0 +1,413 @@
+{
```
I'm not sure about having a random example,json in the root of the project. If you want it for testing purposes, you could just add it to .gitignore?

ZikoGotGame commented on src/scheduler_cli/courses.py:
```diff
@@ -0,0 +1,45 @@
+from .json import JsonConfig
+from scheduler import load_config_from_file, CombinedConfig, SchedulerConfig, CourseConfig
+from typing import Any
+
```
Looks about right. Obviously there should still be documentation, cleaned up spacing, and error checking. I know it is not done yet, so looks good :)

ZikoGotGame commented on src/scheduler_cli/courses.py:
```diff
@@ -0,0 +1,45 @@
+
+
+def add_course(json_config: JsonConfig, course_id: str, credits: int, room: list[str], lab: list[str], conflicts: list[str], faculty: list[str]) -> None:
+
```
Also. make sure these are static methods in a class. I don't want to have to import the function individually.

Tommy-Nguyen17 committed:
updated signatures and implementation of mod and del

Tommy-Nguyen17 committed:
test_courses.py & update to courses.py

Created test_courses.py for unit tests but not entirely sure how to write them

Updated implementation of courses.py
